### PR TITLE
proxy bypass check for specific request #1791

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: test/server/external-proxy-test.js [hunk 1]
```diff
@@ -171,6 +176,18 @@ describe('External proxy', () => {
             });
     });
 
+    it('Should send the http request and bypass the proxy', () => {
```
Comment (miherlosev) [inline test/server/external-proxy-test.js]: Let's exclude tests for this feature to the separate module.
Test structure:
```js
module('Should handle proxy bypass requests', () => {
        test('http');
        test('https');
});
```

File: src/session/index.js [hunk 2]
```diff
@@ -108,16 +108,23 @@ export default class Session extends EventEmitter {
         return taskScript;
     }
 
-    setExternalProxySettings (proxyUrl) {
-        const parsedUrl = typeof proxyUrl === 'string' ? parseUrl('http://' + proxyUrl) : null;
-        let settings    = null;
+    setExternalProxySettings (proxySettings) {
+        if (typeof proxySettings === 'string')
+            proxySettings = { proxyUrl: proxySettings };
```
Comment (miherlosev) [inline src/session/index.js]: Ok. Let it be just `url` instead of `proxyUrl`.

File: test/server/external-proxy-test.js [hunk 3]
```diff
@@ -282,4 +287,42 @@ describe('External proxy', () => {
             }
         });
     });
+
+    describe('Bypass external proxy for specific urls', () => {
```
Comment (miherlosev) [inline test/server/external-proxy-test.js]: Rename as described in previos review comments

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
//cc @LavrovArtem

**Pull request merged**